### ISSUE TITLE
[java] HardcodedCryptoKey ignores method call

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3369]
**Affects PMD Version:6.35.0** 

**Rule:HardcodedCryptoKey**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.35.0/pmd_rules_java_security.html#hardcodedcryptokey>

**Description:**
PMD should have reported a warning to detect the bug because when str is assigned by final variable globalHardcodedKey, the program will exist a security flaw.

**Code Sample demonstrating the issue:**

```java
public static final String globalHardcodedKey = "my scecret key";
public void func() {
  testHardCodedCryptoKey(s)
}
public void testHardCodedCryptoKey(String s) {       
  secretKeySpec = new SecretKeySpec(s.getBytes(), "AES");  // should report a warning here
}
```

**Expected outcome:**

PMD should report a violation at line 6, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Belle-PL** [comment]
@adangel, Hi, Andreas, could you please help me check this issue?
